CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -58,16 +58,14 @@ public enum ChildKind {
 /// A child of a node, that may be declared optional or a token with a
 /// restricted subset of acceptable kinds or texts.
 public class Child {
+  /// If the child has been renamed, its old, now deprecated, name.
+  private let deprecatedName: String?
+
   /// The name of the child.
   ///
   /// The first character of the name is always uppercase.
   public let name: String
```

Comment:
I've been thinking about privatizing the `name` property as proposed in the original discussion but we are using it in the role of `id` to identify children in tests `ValidateSyntaxNodes.swift` for example: https://github.com/apple/swift-syntax/blob/e340af46cd6725e1567c42adcbb1ec979bf7f7b4/CodeGeneration/Tests/ValidateSyntaxNodes/ValidateSyntaxNodes.swift#L83-L85

Reply:
You should be able to do `type.description == other.type.description` instead of `name == other.name` here.

Reply:
Done